src/backend/optimizer/plan/plangroupext.c

```diff
@@ -2256,23 +2255,53 @@ rebuild_append_simple_rel_and_rte(PlannerInfo *root,
 	Assert(list_length(rtable) == list_length(subplans));
 	Assert(list_length(subplans) == list_length(subroots));
 
+	/*
+	 * First of all, build RTE array before initialization RelOptInfos because
+	 * inherited relations requires full list of already initialized child ones.
```

Comment:
Does full statement confuse?

Reply:
It's quite clear, but an English reader would need some effort to correct mistakes.

We could also add a reference to [`build_simple_rel()`](https://github.com/arenadata/gpdb/pull/101/files#diff-6407594701ec72298a72da1ff32b5241R2306), stating that it may be recursive - as this method is the reason why this "extra" cycle is needed.

Reply:
fixed, thx